### ISSUE TITLE
Unable to test gamemode

### ISSUE THREAD

**Issue by ADRFranklin** [Southclaws/samp-plugin-boilerplate#1]
I am using Gitlab CI to build this project, but after building gamemode is not able to find the test gamemode amx and actually run the server, which ends up in a hanging up the CI build process til it times out.

By default my CI Runners use `Ubuntu 16.04` as the image

# gitlab-ci.yml
```
stages:
    - build
    - test
    - release

cache:
    paths:
        - test/

build:
    stage: build
    image: docker:latest
    services:
        - docker:dind       
    script:
        - docker build .

test:
    stage: test
    before_script:
        - apt-get update && apt-get install -y curl wget make sudo git g++-multilib
        - curl https://raw.githubusercontent.com/Southclaws/sampctl/master/install-deb.sh | sh
    script:
        - make test-setup
        - make test-linux
```

# CI Build Error
```
make test-setup[0;m
cd test && sampctl server ensure
INFO: Usage metrics are active. See https://github.com/Southclaws/sampctl/wiki/Usage-Metrics for more information.
updating runtimes list...
INFO: Downloading package 0.3.7 into /builds/SanAndreasGaming/XO/test
sampctl package ensure
Failed to find configuration /root/.samp/config.toml
Failed to find configuration /root/.samp/config.yaml
INFO: michaelfranklin/XO successfully ensured dependency files for github.com/Southclaws/samp-stdlib
INFO: michaelfranklin/XO successfully ensured dependency files for github.com/sampctl/pawn-stdlib
INFO: michaelfranklin/XO successfully ensured dependency files for github.com/pawn-lang/YSI-Includes
INFO: michaelfranklin/XO successfully ensured dependency files for github.com/Zeex/amx_assembly
INFO: ensured dependencies for package
[32;1m$ make test-linux[0;m
sampctl package build
Failed to find configuration /root/.samp/config.toml
Failed to find configuration /root/.samp/config.yaml
updating compiler list...
INFO: Downloading compiler package v3.10.4
INFO: Build  successful with 0 problems
cd test && sampctl server run
Failed to find configuration /root/.samp/config.toml
Failed to find configuration /root/.samp/config.yaml
Script[gamemodes/test.amx]: Run time error 19: "File or function is not found"
Number of vehicle models: 0
```

# Pawn.json
```
{
  "user": "michaelfranklin",
  "repo": "XO",
  "entry": "test.pwn",
  "output": "test/gamemodes/test.amx",
  "dependencies": ["Southclaws/samp-stdlib"],
  "dev_dependencies": ["pawn-lang/YSI-Includes"],
  "runtime": {
    "plugins": ["michaelfranklin/XO"]
  }
}
```

# Samp.json
```
{
  "gamemodes": ["test"],
  "plugins": ["XO"],
  "rcon_password": ";,JoTh856/:?",
  "port": 7777,
  "hostname": "test",
  "maxplayers": 32,
  "mode": "y_testing"
}
```

I am not sure if I have missed a step or have incorrectly attempted this.

# Edit

Some more information, after running it manually using `mkdir build && cmake .. && make` I was able to figure out why it wasn't building correctly, and it gave me this output

```
/builds/SanAndreasGaming/XO/lib/samp-plugin-sdk/amxplugin2.cpp: In function 'std::__cxx11::string amx_GetCppString(AMX*, cell)':
/builds/SanAndreasGaming/XO/lib/samp-plugin-sdk/amxplugin2.cpp:89:15: error: 'nullptr' was not declared in this scope
  cell *addr = nullptr;
               ^
/builds/SanAndreasGaming/XO/lib/samp-plugin-sdk/amxplugin2.cpp: In function 'int amx_SetCppString(AMX*, cell, const string&, size_t)':
/builds/SanAndreasGaming/XO/lib/samp-plugin-sdk/amxplugin2.cpp:103:15: error: 'nullptr' was not declared in this scope
  cell *dest = nullptr;
```

I am using the default files with the exception of changing the name files and include guards but left rest until I was able to get a working build.

# Current Gitlab CI config
```
stages:
    - build
    - test
    - release

cache:
    paths:
        - test/

variables:
  GIT_SUBMODULE_STRATEGY: recursive

before_script:
    - apt-get update && apt-get install -y curl wget make cmake sudo git g++-multilib 

build:
    stage: build     
    script:
        - mkdir build
        - cd build && cmake .. && make

test:
    stage: test
    before_script:
        - curl https://raw.githubusercontent.com/Southclaws/sampctl/master/install-deb.sh | sh
    script:
        - make test-setup
        - make test-linux  
```

# Edit 2

Fixed the conpiling issue by adding `-std=gnu++11` to the `CMAKE_CXX_FLAGS` flags

# CMake file
```
cmake_minimum_required(VERSION 3.0)

project(XO)

list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/lib/cmake-modules")

if(UNIX)
	set(CMAKE_CXX_FLAGS "-m32 -fvisibility=hidden -std=gnu++11")
	set(CMAKE_C_FLAGS "-m32 -fvisibility=hidden")
	set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS OFF)
endif()

# removes the sprintf warnings from plugin-natives
add_definitions(-D_CRT_SECURE_NO_WARNINGS)

set(CMAKE_BINARY_DIR ${CMAKE_SOURCE_DIR}/test/plugins)
set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR})
set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR})

add_subdirectory(src)
```

**Comment by ADRFranklin** [comment]
The output of the build seems to be incorrect

# Bulid Log
```
/builds/SanAndreasGaming/XO/src/main.cpp:19:39: warning: 'nativeList' initialized and declared 'extern'
 extern "C" AMX_NATIVE_INFO nativeList[] = {
                                       ^
[ 71%] Building CXX object src/CMakeFiles/projectname.dir/impl.cpp.o
[ 85%] Building CXX object src/CMakeFiles/projectname.dir/natives.cpp.o
[100%] Linking CXX shared module ../../test/plugins/projectname.so
```

As you can see it is outputting it as `projectname.so` and not what I set it as `XO`

**Comment by ADRFranklin** [comment]
Fixed in PR #2
